Lib/datetime.py

```diff
@@ -454,20 +454,24 @@ def __new__(cls, days=0, seconds=0, microseconds=0,
         return self
 
     def __repr__(self):
-        if self._microseconds:
-            return "%s.%s(%d, %d, %d)" % (self.__class__.__module__,
-                                          self.__class__.__qualname__,
-                                          self._days,
-                                          self._seconds,
-                                          self._microseconds)
+        args = []
+
+        if self._days:
+            args.append("days=%d" % self._days)
+
         if self._seconds:
-            return "%s.%s(%d, %d)" % (self.__class__.__module__,
-                                      self.__class__.__qualname__,
-                                      self._days,
-                                      self._seconds)
-        return "%s.%s(%d)" % (self.__class__.__module__,
+            args.append("seconds=%d" % self._seconds)
+
+        if self._microseconds:
+            args.append("microseconds=%d" % self._microseconds)
+
+        if len(args) == 0:
```

Comment:
Replace this with:

if not args:
   args.append('0')

to prevent code duplication.